autoACDC/autoACDC.py

```diff
@@ -0,0 +1,501 @@
+            raise Exception("RequestType is not 'Resubmission'")
+
+        # WF must be in assignment-approved in order to be assigned
+        if (self.schema["RequestStatus"] != "assignment-approved") and not self.options['testbed_assign']:
+            raise Exception("RequestType is not 'assignment-approved'")
+
+    def getPreviousWfs(self):
+        """
+        Returns: the first (ancestor) and last (original) workflow infos.
+        """
+
+        if 'OriginalRequestName' in self.schema:
+            original_wf = workflowInfo(self.url, self.schema['OriginalRequestName'])            
+            ancestor_wf = workflowInfo(self.url, self.schema['OriginalRequestName'])
+            ## go back as up as possible
+            while ancestor_wf.request['RequestType'] == 'Resubmission':
+                if 'OriginalRequestName' not in ancestor_wf.request:
+                    ancestor_wf = None
+                    break
+                ancestor_wf = workflowInfo(self.url, ancestor_wf.request['OriginalRequestName'])
+        else:
+            raise Exception("'OriginalRequestName' not in schema.")
+
+        return original_wf, ancestor_wf
+
+    def getAssignParameters(self):
+        """
+        Defines a dictionary of paremeters for assign based on the
+        original workflow, the ACDC created, and the options that
+        are passed to the class.
+        Returns: a dictionary of parameters for assign.
+        """
+
+        self.setACDCWfInfo()
+
+        original_wf, ancestor_wf = self.getPreviousWfs()
+
+        # check to see if the workflow is a task chain or an ACDC of a taskchain
+        taskchain = (self.schema["RequestType"] == "TaskChain") or (ancestor_wf and ancestor_wf.request["RequestType"] == "TaskChain")
+
+        # these are automatically determined from ancestor workflow
+        era = ancestor_wf.acquisitionEra()
+        procstring = ancestor_wf.processingString()
+        if (not era or not procstring) or (taskchain and (type(era)!=dict or type(procstring)!=dict)):
+            raise Exception("We do not have a valid AcquisitionEra and ProcessingString")
+        procversion = ancestor_wf.request['ProcessingVersion']
+        
+        # some default options in case the arguments aren't passed
+
+        # Must use --lfn option, otherwise workflow won't be assigned
+        if self.options['lfn']:
+            lfn = self.options['lfn']
+        elif "MergedLFNBase" in self.schema:
+            lfn = self.schema['MergedLFNBase']
+        elif ancestor_wf and "MergedLFNBase" in ancestor_wf.request:
+            lfn = ancestor_wf.request['MergedLFNBase']
+        else:
+            raise Exception("Can't assign the workflow! Please include workflow lfn using --lfn option.")
+        
+        # activity production by default for taskchains, reprocessing for default by workflows
+        if self.options['activity']:
```

Comment:
Let's not make this optional. We should look at the original workflow for `Dashboard` and use whatever it uses. Perhaps, if you don't do anything, that's the default behavior.